### PR TITLE
Removes a bunch of rawtypes warnings

### PR DESCRIPTION
**Pull request by jkff**
Except for Spark and Flink runners and the microbenchmarks module.

### CODE REVIEW

**Comment by jkff** [comment]
R: @dhalperi

**Comment by dhalperi** [comment]
I'm sure you've seen this does not pass yet :)

**Comment by jkff** [comment]
Yeah. I give up because the generated Avro code contains warnings (until they release https://github.com/apache/avro/pull/124 and we update avro plugin) and I couldn't find a way to suppress them, nor to tell javac to omit this warning for that file, nor even to make javac use different parameters to compile tests than to compile the main code.

I'd rather resume after Avro gets fixed.

**Comment by jkff** [comment]
OK, I decided to change this PR to simply reduce the number of these warnings, but not upgrade them to errors, in order to reduce future work when we find a way to cleanly make this an error.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java [hunk 1]
```diff
@@ -2157,25 +2150,18 @@ protected String getKindString() {
...
 
     @Override
-    @SuppressWarnings({"rawtypes", "unchecked"})
     public void translate(
-        StreamingPubsubIOWrite transform,
-        TranslationContext context) {
-      translateTyped(transform, context);
-    }
-
-    private <T> void translateTyped(
```
Comment (dhalperi) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java]: nice

File: sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java [hunk 2]
```diff
@@ -20,6 +20,8 @@
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Preconditions.checkNotNull;
 
+import com.google.common.annotations.VisibleForTesting;
+
```
Comment (dhalperi) [inline sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java]: drop?

Comment (jkff) [inline sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java]: Done.

**Comment by dhalperi** [comment]
LGTM

**Pull request closed without merging**